### DISCUSSION TITLE
Will `jax.lax.while` support dynamic shapes?

### DISCUSSION THREAD

**Discussion by femtomc** [jax-ml/jax#12427] [Q&A]
How would parametric shapes work inside a construct like `jax.lax.while`? What are the restrictions? Is this even something that is intended to be supported?

**Comment by daskol** [comment]
Primitive `while` requires static shapes. As far as I understand, there are some complications with `while` for dynamic shapes it seems that there is not support for dynamic shapes on the low-level \[[1][1]\]\[[2][2]\].

[1]: https://www.tensorflow.org/xla/operation_semantics#while
[2]: https://github.com/openxla/stablehlo/blob/main/docs/spec.md#types
